Validation on DTO should be using @Introspected on a model class not @Valid in Resource class

##### **Overview of the issue**

https://github.com/micronaut-projects/micronaut-core/issues/2303

Micronaut uses `@Introspected` in place of `@Valid` for validation, existing resource integrations tests (with DTOs) are failing:
```
[ERROR]   TopicResourceIT.updateTopic:199 
Expecting:
 <400>
to be equal to:
 <200>
but was not.
```
as validation isn't triggered in `mvn verify`

##### **Reproduce the error**

Generate project with model that has validators, eg. `required` field.

##### **Suggest a Fix**

`@Valid` should be removed from resource method definitions and `@Introspected` put on a DTO.

A negative test should be added to check if such validation is triggered:
```
    @Test
    public void updateTopicWithInvalidTopicName() throws Exception {
        // Initialize the database
        topicRepository.saveAndFlush(topic);

        int databaseSizeBeforeUpdate = topicRepository.findAll().size();

        // Update the topic
        Topic updatedTopic = topicRepository.findById(topic.getId()).get();

        updatedTopic
            .topicName(null) // null is not valid topic name
            .description(UPDATED_DESCRIPTION)
            .link(UPDATED_LINK);

        TopicDTO topicDTO = topicMapper.toDto(updatedTopic);

        HttpResponse<Topic> response = client.exchange(HttpRequest.PUT("/api/topics", topicDTO), Topic.class)
            .onErrorReturn(t -> (HttpResponse<Topic>) ((HttpClientResponseException) t).getResponse()).blockingFirst();

        assertThat(response.status().getCode()).isEqualTo(HttpStatus.BAD_REQUEST.getCode());

        // Validate the Topic in the database
        List<Topic> topicList = topicRepository.findAll();
        assertThat(topicList).hasSize(databaseSizeBeforeUpdate);
        Topic testTopic = topicList.get(topicList.size() - 1);

        assertThat(testTopic.getTopicName()).isEqualTo(DEFAULT_TOPIC_NAME);
        assertThat(testTopic.getDescription()).isEqualTo(DEFAULT_DESCRIPTION);
        assertThat(testTopic.getLink()).isEqualTo(DEFAULT_LINK);
    }
```

--- a/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs
@@ -18,6 +18,7 @@
 -%>
 package <%= packageName %>.service.dto;
 
+import io.micronaut.core.annotation.Introspected;
 <%_
 let { importApiModelProperty, importJsonIgnore, importJsonIgnoreProperties, importSet, uniqueEnums } = _processJavaEntityImports(fields, relationships);
 _%>
@@ -69,6 +70,7 @@ import <%= packageName %>.domain.enumeration.<%= element %>;
 <%_ if (typeof javadoc !== 'undefined') { _%>
 @ApiModel(description = "<%- formatAsApiDescription(javadoc) %>")
 <%_ } _%>
+@Introspected
 public class <%= asDto(entityClass) %> implements Serializable {
 
     private <%= primaryKeyType %> id;

--- a/generators/entity-server/templates/src/main/java/package/web/rest/EntityResource.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/web/rest/EntityResource.java.ejs
@@ -113,9 +113,6 @@ import org.springframework.http.HttpStatus; %>
 import reactor.core.publisher.Flux;
 <%_ } _%>
 import javax.annotation.Nullable;
-<%_ if (validation) { _%>
-import javax.validation.Valid;
-<%_ } _%>
 import java.net.URI;
 import java.net.URISyntaxException;
 <%_ if (pagination === 'no' && dto === 'mapstruct' && !viaService && fieldsContainNoOwnerOneToOne === true) { _%>
@@ -172,7 +169,7 @@ public class <%= entityClass %>Resource {
      */
     @Post("/<%= entityApiUrl %>")
     @ExecuteOn(TaskExecutors.IO)
-    public HttpResponse<<%= instanceType %>> create<%= entityClass %>(<% if (validation) { %>@Valid <% } %>@Body <%= instanceType %> <%= instanceName %>) throws URISyntaxException {
+    public HttpResponse<<%= instanceType %>> create<%= entityClass %>(@Body <%= instanceType %> <%= instanceName %>) throws URISyntaxException {
         log.debug("REST request to save <%= entityClass %> : {}", <%= instanceName %>);
         if (<%= instanceName %>.getId() != null) {
             throw new BadRequestAlertException("A new <%= entityInstance %> cannot already have an ID", ENTITY_NAME, "idexists");
@@ -212,7 +209,7 @@ public class <%= entityClass %>Resource {
      */
     @Put("/<%= entityApiUrl %>")
     @ExecuteOn(TaskExecutors.IO)
-    public HttpResponse<<%= instanceType %>> update<%= entityClass %>(<% if (validation) { %>@Valid <% } %>@Body <%= instanceType %> <%= instanceName %>) throws URISyntaxException {
+    public HttpResponse<<%= instanceType %>> update<%= entityClass %>(@Body <%= instanceType %> <%= instanceName %>) throws URISyntaxException {
         log.debug("REST request to update <%= entityClass %> : {}", <%= instanceName %>);
         if (<%= instanceName %>.getId() == null) {
             throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "idnull");

